src/web/services/course.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { JoinStatus, MessageOutput } from '../types/api-output';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles course related logic provision.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class CourseService {
+
+  constructor(private httpRequestService: HttpRequestService) {
+  }
+
+  /**
+   * Join a course by calling API.
+   */
+  joinCourse(regKey: string, entityType: string): Observable<JoinStatus> {
+    const paramMap: { [key: string]: string } = {
+      key: regKey,
+      entitytype: entityType,
+    };
+    return this.httpRequestService.get('/join', paramMap);
+  }
+
+  /**
+   * Send reminder emails to unregistered students.
```

Comment:
`Send join reminder emails to unregistered students.`.

Same for other method.